TheApiAwakens/StarWarsDetailViewController.swift

```diff
@@ -0,0 +1,551 @@
+            case .character :
+                self.navigationItem.title = ResourceType.character.rawValue
+                fetchForCharacter(with: nextPageNumber)
+            default: break
+        }
+    }
+
+    func hideCharacterLabels() {
+        charactersVehicleStarshipLabel.isHidden = true
+        charactersVehicleStarshipValueLabel.isHidden = true
+    }
+
+    func fetchForStarship(with page: Int) {
+        swAPIClient.fetchForStarship(nextPage: page, completion: { [weak self] (result) in
+            switch result {
+                case .failure(let error):
+                    print(error)
+                case .success(let result):
+                    self?.hasFinishedRequest = true
+                    
+                    // Update pickerView with data
+                    self?.pickerData = [String]()
+                    self?.starships += result.resource
+                    self?.objectQuantity = (self?.starships.count)!
+                    self?.hasNextPage = result.hasPage
+                    for starship in (self?.starships)! {
+                        self?.pickerData.append(starship.name)
+                    }
+                    self?.pickerView.reloadAllComponents()
+                    
+                    //Set labels with first element of the array since the user has not selected any row of the pickerView yet
+                    if (self?.isApiFirstCall)! {
+                        self?.setLabels(with: (self?.starships.first!)!)
+                        self?.isApiFirstCall = false
+                    }
+            }
+        })
+    }
+
+    func fetchForVehicle(with page: Int) {
+        swAPIClient.fetchForVehicle(nextPage: page, completion: { [weak self] (result) in
+            switch result {
+                case .failure(let error):
+                    print(error)
+                case .success(let result):
+                    self?.hasFinishedRequest = true
+                    
+                    // Update pickerView with data
+                    self?.pickerData = [String]()
+                    self?.vehicles += result.resource
+                    self?.objectQuantity = (self?.vehicles.count)!
+                    self?.hasNextPage = result.hasPage
+
+                    for vehicle in (self?.vehicles)! {
+                    self?.pickerData.append(vehicle.name)
+                }
+                self?.pickerView.reloadAllComponents()
+                
+                //Set labels with first element of the array since the user has not selected any row on the pickerView yet
+                if (self?.isApiFirstCall)! {
+                    self?.setLabels(with: (self?.vehicles.first!)!)
+                    self?.isApiFirstCall = false
+                }
+            }
+        })
+    }
+    //Set labels only for Vehcile and Starship (Character is not a TransportCraft)
+    func setLabels(with valueSelected: TransportCraft) {
```

Comment:
**Food for thought**:  Do you need to handle starships and vehicles in different ways? Can't you always treat them as `TransportCraft`?